api/controllers/auth.js

```diff
@@ -44,6 +50,48 @@ const registration = async (req, res, next) => {
   }
 };
 
+const reset = async (req, res, next) => {
+  const { email } = req.body;
+  try {
+    const v = await emailSchema.validate({ email });
+    const a = v && (await services.resetPasswordRequest({ email }));
+    let status = a.status ? a.status : 200;
+    let response = a.messageId ? 'Mail was sent' : a.message;
+    res.status(status).send(response);
+  } catch (e) {
+    next(e);
```

Comment:
what do you know about middlewares? We already have error handler middleware. All throws will be there and you can do anything you want there

Reply:
The new errorHandling is must to move in new version of API(v1, v2, etc) not to broke the working code. The news about code contracts must be posted in WIKI and into Skype chat.